diff --git a/node_modules/react-native/React-Core-prebuilt.podspec b/node_modules/react-native/React-Core-prebuilt.podspec
index 98aa6a0..e3bbaa7 100644
--- a/node_modules/react-native/React-Core-prebuilt.podspec
+++ b/node_modules/react-native/React-Core-prebuilt.podspec
@@ -7,7 +7,7 @@ require "json"
 package = JSON.parse(File.read(File.join(__dir__, "package.json")))
 version = package['version']
 
-source = ReactNativeCoreUtils.resolve_podspec_source()
+source = ReactNativeCoreUtils.resolve_podspec_source() || { :git => 'https://github.com/facebook/react-native.git', :tag => "v#{version}" }
 Pod::Spec.new do |s|
   s.name                   = "React-Core-prebuilt"
   s.version                = version
diff --git a/node_modules/react-native/scripts/cocoapods/helpers.rb b/node_modules/react-native/scripts/cocoapods/helpers.rb
index cdb1e14..fbd2539 100644
--- a/node_modules/react-native/scripts/cocoapods/helpers.rb
+++ b/node_modules/react-native/scripts/cocoapods/helpers.rb
@@ -31,7 +31,7 @@ class Finder
     def self.find_codegen_file(path)
         js_files = '-name "Native*.js" -or -name "*NativeComponent.js"'
         ts_files = '-name "Native*.ts" -or -name "*NativeComponent.ts"'
-        return `find #{path} -type f \\( #{js_files} -or #{ts_files} \\)`.split("\n").sort()
+        return `find "#{path}" -type f \\( #{js_files} -or #{ts_files} \\)`.split("\n").sort()
     end
 end
 
diff --git a/node_modules/react-native/scripts/cocoapods/new_architecture.rb b/node_modules/react-native/scripts/cocoapods/new_architecture.rb
index 3873632..f6e4cf0 100644
--- a/node_modules/react-native/scripts/cocoapods/new_architecture.rb
+++ b/node_modules/react-native/scripts/cocoapods/new_architecture.rb
@@ -171,7 +171,7 @@ class NewArchitectureHelper
         excluded_info_plist = ["/Pods", "Tests", "metainternal", ".bundle", "build/", "DerivedData/", ".xcframework", ".framework", "watchkitapp", "today-extention"]
         projectPaths.each do |projectPath|
             projectFolderPath = File.dirname(projectPath)
-            infoPlistFiles = `find #{projectFolderPath} -name "Info.plist"`
+            infoPlistFiles = `find "#{projectFolderPath}" -name "Info.plist"`
             infoPlistFiles = infoPlistFiles.split("\n").map { |f| f.strip }
 
             infoPlistFiles.each do |infoPlistFile|
diff --git a/node_modules/react-native/scripts/cocoapods/rncore.rb b/node_modules/react-native/scripts/cocoapods/rncore.rb
index 09cd883..66529ca 100644
--- a/node_modules/react-native/scripts/cocoapods/rncore.rb
+++ b/node_modules/react-native/scripts/cocoapods/rncore.rb
@@ -160,7 +160,7 @@ class ReactNativeCoreUtils
         rncore_log("  #{Pathname.new(destinationDebug).relative_path_from(Pathname.pwd).to_s}")
         rncore_log("  #{Pathname.new(destinationRelease).relative_path_from(Pathname.pwd).to_s}")
 
-        return {:http => URI::File.build(path: destinationDebug).to_s }
+        return {:http => 'file://' + URI::Parser.new.escape(destinationDebug) }
     end
 
     def self.podspec_source_download_prebuilt_nightly_tarball()
@@ -196,7 +196,7 @@ class ReactNativeCoreUtils
         rncore_log("Resolved nightly ReactNativeCore-prebuilt version:")
         rncore_log("  #{Pathname.new(destinationDebug).relative_path_from(Pathname.pwd).to_s}")
         rncore_log("  #{Pathname.new(destinationRelease).relative_path_from(Pathname.pwd).to_s}")
-        return {:http => URI::File.build(path: destinationDebug).to_s }
+        return {:http => 'file://' + URI::Parser.new.escape(destinationDebug) }
     end
 
     def self.process_dsyms(frameworkTarball, dSymsTarball)
diff --git a/node_modules/react-native/scripts/cocoapods/rndependencies.rb b/node_modules/react-native/scripts/cocoapods/rndependencies.rb
index a901840..2982c90 100644
--- a/node_modules/react-native/scripts/cocoapods/rndependencies.rb
+++ b/node_modules/react-native/scripts/cocoapods/rndependencies.rb
@@ -161,7 +161,7 @@ class ReactNativeDependenciesUtils
         destinationDebug = download_stable_rndeps(@@react_native_path, @@react_native_version, :debug)
         download_stable_rndeps(@@react_native_path, @@react_native_version, :release)
 
-        return {:http => URI::File.build(path: destinationDebug).to_s }
+        return {:http => 'file://' + URI::Parser.new.escape(destinationDebug) }
     end
 
     def self.release_tarball_url(version, build_type)
@@ -228,7 +228,7 @@ class ReactNativeDependenciesUtils
         destinationDebug = download_nightly_rndeps(@@react_native_path, @@react_native_version, :debug)
         download_nightly_rndeps(@@react_native_path, @@react_native_version, :release)
 
-        return {:http => URI::File.build(path: destinationDebug).to_s }
+        return {:http => 'file://' + URI::Parser.new.escape(destinationDebug) }
         return {:http => url}
     end
 
diff --git a/node_modules/react-native/third-party-podspecs/ReactNativeDependencies.podspec b/node_modules/react-native/third-party-podspecs/ReactNativeDependencies.podspec
index e08e6c2..fa51281 100644
--- a/node_modules/react-native/third-party-podspecs/ReactNativeDependencies.podspec
+++ b/node_modules/react-native/third-party-podspecs/ReactNativeDependencies.podspec
@@ -21,7 +21,7 @@ end
 package = JSON.parse(File.read(File.join(react_native_path, "package.json")))
 version = package['version']
 
-source = ReactNativeDependenciesUtils.resolve_podspec_source()
+source = ReactNativeDependenciesUtils.resolve_podspec_source() || { :git => 'https://github.com/facebook/react-native.git', :tag => "v#{version}" }
 
 Pod::Spec.new do |spec|
   spec.name                 = 'ReactNativeDependencies'
